drivers/serial/uart_infineon_cat1.c

```diff
@@ -0,0 +1,506 @@
+		parity = CYHAL_UART_PARITY_ODD;
+		break;
+	case UART_CFG_PARITY_EVEN:
+		parity = CYHAL_UART_PARITY_EVEN;
+		break;
+	default:
+		parity = CYHAL_UART_PARITY_NONE;
+	}
+	return (parity);
+}
+
+static uint32_t _convert_uart_stop_bits_z_to_cyhal(uint8_t z_stop_bits)
+{
+	uint32_t stop_bits;
+
+	switch ((enum uart_config_stop_bits) z_stop_bits) {
+	case UART_CFG_STOP_BITS_1:
+		stop_bits = 1u;
+		break;
+
+	case UART_CFG_STOP_BITS_2:
+		stop_bits = 2u;
+		break;
+	default:
+		stop_bits = 1u;
+	}
+	return (stop_bits);
+}
+
+static uint32_t _convert_uart_data_bits_z_to_cyhal(uint8_t z_data_bits)
+{
+	uint32_t data_bits;
+
+	switch ((enum uart_config_data_bits) z_data_bits) {
+	case UART_CFG_DATA_BITS_5:
+		data_bits = 1u;
+		break;
+
+	case UART_CFG_DATA_BITS_6:
+		data_bits = 6u;
+		break;
+
+	case UART_CFG_DATA_BITS_7:
+		data_bits = 7u;
+		break;
+
+	case UART_CFG_DATA_BITS_8:
+		data_bits = 8u;
+		break;
+
+	case UART_CFG_DATA_BITS_9:
+		data_bits = 9u;
+		break;
+
+	default:
+		data_bits = 1u;
+	}
+	return (data_bits);
+}
+
+
+/* UART CONSOLE AND CONFIGURATION API */
+static int uart_cat1_poll_in(const struct device *dev, unsigned char *c)
+{
+	cy_rslt_t rec;
+
+	rec = cyhal_uart_getc(&DEV_DATA(dev)->obj, c, 0u);
+
+	return ((rec == CY_SCB_UART_RX_NO_DATA) ? -1 : 0);
+}
+
+static void uart_cat1_poll_out(const struct device *dev, unsigned char c)
+{
+	(void) cyhal_uart_putc(&DEV_DATA(dev)->obj, (uint32_t)c);
+}
+
+static int uart_cat1_err_check(const struct device *dev)
+{
+	uint32_t status = Cy_SCB_UART_GetRxFifoStatus(DEV_DATA(dev)->obj.base);
+	int errors = 0;
+
+	if (status & CY_SCB_UART_RX_OVERFLOW) {
+		errors |= UART_ERROR_OVERRUN;
+	}
+
+	if (status & CY_SCB_UART_RX_ERR_PARITY) {
+		errors |= UART_ERROR_PARITY;
+	}
+
+	if (status & CY_SCB_UART_RX_ERR_FRAME) {
+		errors |= UART_ERROR_FRAMING;
+	}
+
+	return errors;
+}
+
+static int uart_cat1_configure(const struct device *dev,
+			       const struct uart_config *cfg)
+{
+	cy_rslt_t result;
+	struct uart_cat1_data *data = DEV_DATA(dev);
+
+	bool enable_cts = (data->obj.pin_cts != NC);
+	bool enable_rts = (data->obj.pin_rts != NC);
+
+	if (cfg == NULL) {
+		return -EINVAL;
+	}
+
+	cyhal_uart_cfg_t uart_cfg = {
+		.data_bits = _convert_uart_data_bits_z_to_cyhal(cfg->data_bits),
+		.stop_bits = _convert_uart_stop_bits_z_to_cyhal(cfg->stop_bits),
+		.parity = _convert_uart_parity_z_to_cyhal(cfg->parity)
+	};
+
+	/* Store Uart Zephyr configuration (uart config) into data structure */
+	data->cfg = *cfg;
+
+	/* Configure parity, data and stop bits */
+	result = cyhal_uart_configure(&data->obj, &uart_cfg);
+
+	/* Configure the baud rate */
+	if (result == CY_RSLT_SUCCESS) {
+		result = cyhal_uart_set_baud(&data->obj, cfg->baudrate, NULL);
+	}
+
+	/* Enable flow control */
+	if ((result == CY_RSLT_SUCCESS) && (enable_cts || enable_rts)) {
+		result = cyhal_uart_enable_flow_control(&data->obj, enable_cts,
+							enable_rts);
+	}
+	return (result == CY_RSLT_SUCCESS) ? 0 : -1;
```

Comment:
Please, use errno constants to return errors.

Reply:
updated